### PR TITLE
Implement image grouping via visit numbers

### PR DESCRIPTION
**Pull request by haticekaratay**
This pull request implements a `group_by_visit` function and adds unit tests.

### CODE REVIEW

File: slitlessutils/core/preprocess/crrej/group.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    unique_visits, indices, counts = np.unique(
+        visits, return_inverse=True, return_counts=True
+    )
+    rev = np.split(np.argsort(indices), np.cumsum(counts[:-1]))
+
+    grouped_files = []
+    for r in rev:
+        grouped_files.append(list(files[r]))
+        if len(r) == 1:
+            LOGGER.warning("It's a singleton. The file: {}".format(files[r]))
```
Comment (rosteen) [inline slitlessutils/core/preprocess/crrej/group.py:47]:
```suggestion
            LOGGER.warning(f"The file {files[r][0]} was not grouped with any others."))
```

File: slitlessutils/core/preprocess/crrej/group.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+def group_by_visit(files, return_unique_visits=False, **kwargs):
+    """
+    Group a list of files based on their unique visits.
+
+    Parameters
+    ----------
+    files : list of str
+        List of filenames
```
Comment (rosteen) [inline slitlessutils/core/preprocess/crrej/group.py:14]:
```suggestion
        List of input filenames to be grouped for cosmic ray rejection.
```
Maybe this suggestion is too specific - would this function be used for anything else?

File: slitlessutils/core/preprocess/crrej/group.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    return_unique_visits : bool, optional
+        Indicating whether to return unique visits along with grouped files.
+        If `True`, a tuple containing grouped files and unique visits is returned.
+        If `False`, only grouped files are returned. Default is `False`.
```
Comment (rosteen) [inline slitlessutils/core/preprocess/crrej/group.py:19]:
```suggestion
        Indicates whether to return unique visits along with grouped files.
        If `True`, a list containing lists of grouped files and an array of unique
        visits are returned.
        If `False`, only grouped files are returned. Default is `False`.
```
I think this is more accurate to the data types returned.

File: slitlessutils/core/preprocess/crrej/group.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import numpy as np
+
+from slitlessutils.core.wfss import WFSSCollection
+from ....logger import LOGGER
```
Comment (rosteen) [inline slitlessutils/core/preprocess/crrej/group.py:4]:
```suggestion
from slitless_utils import LOGGER
```
I think this or `from slitlessutils.logger import LOGGER` should work and avoid all the dots.

**Review by rosteen — COMMENTED** [review]
I think the code looks good for the most part, most of my suggestions are wording changes.

**Comment by rosteen** [comment]
One more thing to think about - there's already a `group.py` file in `core/modules/group/` - do we want this file to have a more specific name to avoid confusion/conflict?

**Comment by haticekaratay** [comment]
Thanks @rosteen. Certainly, @Russell-Ryan informed me of this issue offline. As a solution, I will transfer this code to drizzle.py once this [PR](https://github.com/spacetelescope/slitlessutils/pull/27) is merged.

File: slitlessutils/core/preprocess/crrej/group.py (hunk 2 above)
Comment (haticekaratay) [reply slitlessutils/core/preprocess/crrej/group.py:14]: I'm not entirely sure. @Russell-Ryan would have a better answer for this one.

File: slitlessutils/core/preprocess/crrej/group.py (hunk 4 above)
Comment (Russell-Ryan) [reply slitlessutils/core/preprocess/crrej/group.py:4]: I think it should not have an underscore?  `from slitlessutils import LOGGER` ?

File: slitlessutils/core/preprocess/crrej/group.py (hunk 2 above)
Comment (Russell-Ryan) [reply slitlessutils/core/preprocess/crrej/group.py:14]: No, I think the description is ok.  I can't think of any other use case for this code.

**Review by Russell-Ryan — CHANGES_REQUESTED** [review]

File: slitlessutils/core/preprocess/crrej/group.py (hunk 4 above)
Comment (rosteen) [reply slitlessutils/core/preprocess/crrej/group.py:4]: Whoops, good catch - got it right in one spot and not the other.

File: slitlessutils/core/preprocess/crrej/group.py (hunk 2 above)
Comment (rosteen) [reply slitlessutils/core/preprocess/crrej/group.py:14]: Cool, ignore this suggestion then.

**Pull request merged**